[ENG-2045] Add job_id header to prime env eval command

AmeenP opened the pull request:
**Linear:** https://linear.app/primeintellect/issue/ENG-2045

## Summary
Generate and pass job_id header in the prime-cli env eval command to enable end-to-end tracing of evaluation runs.

## Implementation
- Generate unique job_id for each eval run from environment + model + timestamp + uuid
- Pass job_id as header to vf-eval via `--header` flag
- Display job_id to user for reference

## Benefits
- Each eval run gets a unique job_id for tracing
- All requests within an eval run share the same job_id
- job_id is automatically forwarded to inference API for log correlation
- Works seamlessly with pi-inference tracking (ENG-2044)

## Usage
```bash
prime env eval gsm8k -m meta-llama/llama-3.1-70b-instruct
# Output: Eval job_id: gsm8k_meta_llama_llama_3_1_70b_instruct_20251010_120000_abc12345
```

## Changes
- Add uuid import for job ID generation
- Generate unique job_id for each eval run
- Pass job_id as `--header` flag to vf-eval
- Display job_id to user

## Related
- pi-inference PR: https://github.com/PrimeIntellect-ai/pi-inference/pull/36
- pi-inference ticket: ENG-2044

## Test plan
- [x] Syntax validation
- [x] Manual testing with eval command
- [x] Verify job_id appears in inference API logs

samsja commented on src/zeroband/comms.py:
```diff
@@ -159,6 +159,9 @@ def _create_global_pg(self):
 
     def _optimize_ring_ranks(self):
+        self._global_ids = [
+            self.global_store.get(f"gid_{i}").decode("utf-8") for i in range(self.world_info.global_world_size)
```
```suggestion
            self.global_store.get(f"grid_{i}").decode("utf-8") for i in range(self.world_info.global_world_size)
```

typo ?

Jackmin801 replied:
gid = global id. no type :p

samsja replied:
arf I was so proud to have catch a typo

AmeenP committed:
[ENG-2045] Add session_id and job_id headers to prime env eval command

Generate and pass session_id and job_id tracking headers when running
eval commands to enable end-to-end tracing of evaluation runs.

Changes:
- Add uuid import for session ID generation
- Generate unique session_id for each eval run
- Generate job_id from environment + model + timestamp
- Pass both IDs as --header flags to vf-eval
- Display tracking IDs to user for reference

Benefits:
- Each eval run gets unique session_id for tracing
- Related eval runs grouped by job_id
- IDs automatically forwarded to inference API
- Works with pi-inference tracking (ENG-2044)

Related: pi-inference#36

AmeenP committed:
Remove session_id, use only job_id for eval tracking

AmeenP commented:
✅ **Tested locally**

Successfully verified job_id generation and forwarding:

**Test eval run:**
- Command: `prime env eval aime2024 -m meta-llama/llama-3.1-70b-instruct`
- Generated job_id: `aime2024_meta_llama_llama_3.1_70b_instruct_20251010_074236_98be6dbb`
- 18 inference requests all tagged with the same job_id
- All requests visible in pi-inference logs with job_id field

**Verified:**
- ✅ job_id generated correctly from env + model + timestamp + uuid
- ✅ Header `X-PI-Job-Id` passed to vf-eval
- ✅ vf-eval forwards header to inference API
- ✅ job_id appears in all inference API logs
- ✅ End-to-end tracing working perfectly
